src/main/kotlin/no/nav/familie/oppdrag/simulering/SimulerBeregningResponseMapper.kt

```diff
@@ -0,0 +1,61 @@
+package no.nav.familie.oppdrag.simulering
+
+import no.nav.familie.kontrakter.felles.oppdrag.RestSimulerResultat
+import no.nav.familie.oppdrag.rest.SimuleringController
+import no.nav.system.os.entiteter.beregningskjema.BeregningsPeriode
+import no.nav.system.os.tjenester.simulerfpservice.simulerfpservicegrensesnitt.SimulerBeregningResponse
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+@Component
+class SimulerBeregningResponseMapper {
+
+    fun toRestSimulerResult(simulerBeregningResponse: SimulerBeregningResponse,
+                            dato: LocalDate = LocalDate.now()): RestSimulerResultat {
+
+        val totalEtterbetalingsBeløp =
+                simulerBeregningResponse.response?.simulering?.beregningsPeriode?.sumBy { finnEtterbetalingPerPeriode(it, dato) }
+                ?: 0
+
+        return RestSimulerResultat(etterbetaling = totalEtterbetalingsBeløp)
+    }
+
+    private fun finnEtterbetalingPerPeriode(beregningsPeriode: BeregningsPeriode, dato: LocalDate): Int {
+        LOG.info("Finn etterbetalingsperiode for periode: ${beregningsPeriode.periodeFom} ")
+
+        // Fremtidige perioder gir ingen etterbetaling.
+        val datoFraPeriode = LocalDate.parse(beregningsPeriode.periodeFom, DateTimeFormatter.ISO_DATE)
+        if (datoFraPeriode.month > dato.month) return 0
+
+        val stoppNivaBA =
+                beregningsPeriode.beregningStoppnivaa.filter { it.kodeFagomraade?.trim() == "BA" }
+
+        // Feilutbetaling medfører at etterbetaling er 0
+        val inneholderFeilUtbType =
```

Comment:
Betyr denne `inneholder feilutbetaling type`? Isåfall stemmer jeg for å skrive ut variablene.